src/Atoms/UserProfileData/UserProfileData.js

```diff
@@ -0,0 +1,23 @@
+import propTypes from 'prop-types';
+
+import UserProfileDataStyled from './UserProfileDataStyled';
+import participants from '../../Data/MongoDBMock/participants';
+
+const UserProfileData = ({ userData }) => {
+  const userObject = participants.find(
+    (participant) => participant.participantId === userData.userId,
+  );
+  return (
+    <UserProfileDataStyled>
+      <h3>{`${userData.userName} ${userData.userSurname}`}</h3>
```

Comment:
I think this is the place where you really need context data. Use it in this component then, and if you need to fetch more data, most probably you should do it here in useEffect, based on the information you get from context

Reply:
Thanx for this comment. Refactored

Reply:
Context data gathered